docs/decisions/arch-diagrams/mobile-create-order.mmd

```diff
@@ -0,0 +1,72 @@
+sequenceDiagram
+    participant User
+    participant MobileApp
+    participant Backend
+    participant TokenChecker
+    participant GoogleApple
+    participant Commercetools
+    participant EventSystem
+
+    User->>MobileApp: Starts in-app purchase
+    MobileApp->>Backend: Sends purchase info (price, course, platform, token)
+
+    Backend->>Commercetools: Check if user already has a cart
+    Commercetools-->>Backend: Return existing cart or none
+    alt Cart exists
+        Backend->>Commercetools: Delete old cart
+    end
+
+    Backend->>Commercetools: Create a new cart with course and price
```

Comment:
```suggestion
    Backend->>Commercetools: Add course and external price (in localized currency) to a new cart
```

Add: Backend->>Commercetools: Updates cart to set custom field `mobileOrder` to true